src/reconcile/rec_visibility.c

```diff
@@ -116,14 +120,19 @@ __rec_append_orig_value(
      * the tombstone to tell us there is no value between 10 and 20.
      */
     if (unpack->stop_ts != WT_TS_MAX || unpack->stop_txn != WT_TXN_MAX) {
-        WT_ERR(__wt_upd_alloc_tombstone(session, &tombstone, &size));
-        total_size += size;
-        tombstone->txnid = unpack->stop_txn;
-        tombstone->start_ts = unpack->stop_ts;
-        tombstone->durable_ts = unpack->durable_stop_ts;
-
-        tombstone->next = append;
-        append = tombstone;
+        /* No need to append the tombstone if it is already in the update chain. */
+        if (oldest_upd->type != WT_UPDATE_TOMBSTONE) {
+            WT_ERR(__wt_upd_alloc_tombstone(session, &tombstone, &size));
+            total_size += size;
+            tombstone->txnid = unpack->stop_txn;
+            tombstone->start_ts = unpack->stop_ts;
+            tombstone->durable_ts = unpack->durable_stop_ts;
+
+            tombstone->next = append;
+            append = tombstone;
+        } else
+            WT_ASSERT(session,
```

Comment:
How about doing a similar check comparison with `stop_ts` similar like `start_ts` can avoid these changes and solve the problem?
```
if (unpack->stop_ts == upd->start_ts && unpack->stop_txn == upd->txnid &&
          upd->type == WT_UPDATE_TOMBSTONE)
     return (0);
```

Reply:
We can't do this because we may still need to append the onpage value as we only have the tombstone on the update chain.